EIPS/eip-6785.md

```diff
@@ -0,0 +1,205 @@
+    /// f96090b9
+    function utilityHistoryOf(uint256 tokenId) external view returns (string[] memory);
+}
+```
+
+All functions defined as view MAY be implemented as pure or view
+
+Function `setUtilityUri` MAY be implemented as public or external
+
+The event `UpdateUtility` MUST be emitted when the `setUtilityUri` function is called or any other time that the utility
+of the token is changed, like in batch updates. 
+
+The `supportsInterface` method MUST return true when called with `ed231d73`
+
+The original metadata SHOULD conform to the “ERC-6785 Metadata with utilities JSON Schema” which is a compatible
+extension of the “ERC-721 Metadata JSON Schema” defined in `ERC-721`.
+
+“ERC-6785 Metadata with utilities JSON Schema” :
+
+```json
+{
+  "title": "Asset Metadata",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string",
+      "description": "Identifies the asset to which this NFT represents"
+    },
+    "description": {
+      "type": "string",
+      "description": "Describes the asset to which this NFT represents"
+    },
+    "image": {
+      "type": "string",
+      "description": "A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive."
+    },
+    "utilities": {
+      "type": "object",
+      "required": [
+        "type",
+        "description",
+        "t&c"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Describes what type of utility this is"
+        },
+        "description": {
+          "type": "string",
+          "description": "A brief description of the utility"
+        },
+        "properties": {
+          "type": "array",
+          "description": "An array of possible properties describing the utility, defined as key-value pairs",
+          "items": {
+            "type": "object"
+          }
+        },
+        "expiry": {
+          "type": "number",
+          "description": "The period of time for the validity of the utility, since the minting of the NFT. Expressed in seconds"
+        },
+        "t&c": {
+          "type": "string",
+          "description": ""
+        }
+      }
+    }
+  }
+}
+```
+
+## Rationale
+
+Instructions on how to access the utility should be available to the owner of an NFT, as it reduces the dependency on
+third-party platforms and the possibility for fraud because the `utilityUri` containing the instructions can only come
+from the entity that provides or facilitates the utility. Thus, the ability to set the `utilityUri` should be restricted
+to the creator who's offering the utility. Since the `utilityUri` contains information that could be restricted to some
+level and is dependent on a third party platform, the creator needs the ability to modify it in the event the platform
+becomes unavailable or inaccessible.
```

Comment:
Some of this paragraph seems to be adding requirements (sentences containing "should", "may", etc.) Requirements on implementations should go in the specification section.

The rationale section should explain why you made certain technical decisions within the EIP itself.